app/services/aptrust/aptrust_service.rb

```diff
@@ -117,9 +134,12 @@ def self.aptrust_info( work: )
       AptrustInfo.new( work: work ).build
     end
 
-    def self.aptrust_info_write( bag:, work: )
+    def self.aptrust_info_write( bag:, work:, additional_tag_files: )
       aptrust_info = aptrust_info( work: work )
-      File.write( File.join( bag.bag_dir, BAG_FILE_APTRUST_INFO ), aptrust_info, mode: 'w' )
+      file = File.join( bag.bag_dir, BAG_FILE_APTRUST_INFO )
+      File.write( file, aptrust_info, mode: 'w' )
+      # this does not work: bag.tag_files << file
+      additional_tag_files << file
```

Comment:
Hi @fritzfreiheit , did you try this? https://github.com/tipr/bagit/blob/4a7fb6d25218f44f7c1ead5acdb9525ec54f96c7/lib/bagit/manifest.rb#L116

Reply:
@ssciolla -- I looked at that method and it doesn't look like it adds metadata/tag files. It seems to only generate the tags manifest file without adding a new file to the tags manifest file.

Reply:
Hm, seems like there's logic in there for writing the file itself, but if you found it not doing what you want, that's okay too. There does seem to be some weird stuff going on that will make it not work or cause problems if you call it in the wrong order with other methods.